README.md

```diff
@@ -15,6 +15,118 @@ For more information about Interledger specifications that underpin this library
 ## Usage
 For more detail about how to use this library, consult the [wiki](https://github.com/interledger/java-ilp-core/wiki)
 
+The library offers interfaces for all core objects and default builders that produce immutable instances of these objects.
+
+The library also offers a codecs for encoding and decoding the objects according to the ASN.1 specifications and OER (Octet encoding rules).
+
+
+### Interledger Address
+
+Interledger Addresses can be created from a String and then manipulated using utility functions
+
+```java
+//Create a new address (short form)
+InterledgerAddress address = InterledgerAddress.of("private.bob");
+
+//Create a new address (long form)
+InterledgerAddress destinationAddress = InterledgerAddress.builder()
+  .value("private.bob")
+  .build();
+
+//Check if an address starts with a specific prefix
+if( address.startsWith("test") ) {
+  //Testnet address...
+}
+
+```
+
+### Conditions and Fulfillments
+
+Conditions can be derived from a fullfillment or directly from the underlying hash
```

Comment:
We should put a link to the crypto-conditions project and to the PSK rfc so that people can reference what these things are.